src/pages/commits/ActiveCommits/buildVariantCard/WaterfallTaskStatusIcon.tsx

```diff
@@ -0,0 +1,88 @@
+import { useLazyQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import IconButton from "@leafygreen-ui/icon-button";
+import Tooltip from "@leafygreen-ui/tooltip";
+import { Body } from "@leafygreen-ui/typography";
+import { Skeleton } from "antd";
+import { Link } from "react-router-dom";
+import { TaskStatusIcon } from "components/TaskStatusIcon";
+import { getTaskRoute } from "constants/routes";
+import {
+  GetFailedTaskStatusIconTooltipQuery,
+  GetFailedTaskStatusIconTooltipQueryVariables,
+} from "gql/generated/types";
+import { GET_FAILED_TASK_STATUS_ICON_TOOLTIP } from "gql/queries";
+import { isFailedTaskStatus } from "utils/statuses";
+import { msToDuration } from "utils/string";
+
+interface WaterfallTaskStatusIconProps {
+  taskId: string;
+  status: string;
+  displayName: string;
+  timeTaken?: number;
+}
+
+export const WaterfallTaskStatusIcon: React.FC<WaterfallTaskStatusIconProps> = ({
+  taskId,
+  status,
+  displayName,
+  timeTaken,
+}) => {
+  const [loadData, { data, loading }] = useLazyQuery<
+    GetFailedTaskStatusIconTooltipQuery,
+    GetFailedTaskStatusIconTooltipQueryVariables
+  >(GET_FAILED_TASK_STATUS_ICON_TOOLTIP, { variables: { taskId } });
+  const { testResults } = data?.taskTests ?? {};
+  const loadDataCb = () => {
+    if (isFailedTaskStatus(status)) {
```

Comment:
nit. Add a comment saying we only want to check for test results if there are failed tests. For future reference.